sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -284,6 +300,15 @@ private void createRunnerAndConsumersForPTransformRecursively(
         startFunction.run();
       }
 
+      // TODO(BEAM-5972). Make sure that the apiServiceDescriptor is the same.
+      // throughout all references in the InstructionRequest
+      if (apiServiceDescriptor != null) {
```

Comment:
I would think this would always be true for a real bundle, is this check only here because some of our unit tests don't construct valid ProcessBundleDescriptors?

If yes, it may be better to fix those unit tests instead so that we don't introduce a bug later on.

Reply:
See the new code now, I no longer need to lookup the apiServiceDescriptor.